glusterd2/snapshot/lvm/lvm.go

```diff
@@ -0,0 +1,162 @@
+package lvm
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/gluster/glusterd2/glusterd2/brick"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/pborman/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	//CreateCommand is path to lvm create
+	CreateCommand = "/sbin/lvcreate"
+	//RemoveCommand is path to lvm create
+	RemoveCommand = "/sbin/lvremove"
+)
+
+// CommonPrevalidation checks the lvm related validation for snapshot
+func CommonPrevalidation(lvmCommand string) error {
+	fileInfo, err := os.Stat(lvmCommand)
+	if err != nil {
+		return err
+	}
+	switch mode := fileInfo.Mode(); {
+
+	case mode.IsRegular() == false:
+		return err
+	case mode&0111 == 0:
+		return err
+	}
+	return nil
+}
+
+//IsThinLV check for lvm comptability for a path
+func IsThinLV(brickPath string) bool {
+	mountRoot, err := volume.GetBrickMountRoot(brickPath)
+	if err != nil {
+		return false
+	}
+
+	mntInfo, err := volume.GetBrickMountInfo(mountRoot)
+	if err != nil {
+		return false
+	}
+
+	out, err := exec.Command("/sbin/lvs", "--noheadings", "-o", "pool_lv", mntInfo.FsName).Output()
+	if err != nil {
+		return false
+	}
+
+	thinLV := strings.TrimSpace(string(out))
+
+	if thinLV == "" {
+		return false
+	}
+	return true
+}
+
+//MountSnapshotDirectory will mount the snapshot bricks to the given path
+func MountSnapshotDirectory(mountPath string, mountData brick.MountInfo) error {
+	_, err := exec.Command("mount", "-o", mountData.MntOpts, mountData.DevicePath, mountPath).Output()
+	/*
+		logrus.WithFields(logrus.Fields{
+			"device path": b.DevicePath,
+			"mount path":  mountPath,
+			"fs type":     b.FsType,
+			"options":     b.MntOpts,
+		}).Debug("Mounting the device")
+
+		TODO use system mount command to mount the brick
+		err := syscall.Mount(b.DevicePath, mountPath, b.FsType,, syscall.MS_MGC_VAL, b.MntOpts)
+	*/
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//GetVgName creates the device path for lvm snapshot
+func GetVgName(mountDevice string) (string, error) {
+
+	out, err := exec.Command("/sbin/lvs", "--noheadings", "-o", "vg_name", mountDevice).Output()
+	if err != nil {
+		return "", err
+	}
+
+	volGroup := strings.TrimSpace(string(out))
+	return volGroup, nil
+}
+
+//RemoveBrickSnapshot removes an lvm of a brick
+func RemoveBrickSnapshot(mountData brick.MountInfo) error {
+	_, err := exec.Command(RemoveCommand, "f", mountData.DevicePath).Output()
+	return err
+}
+
+//BrickSnapshot takes lvm snapshot of a brick
+func BrickSnapshot(mountData brick.MountInfo, path string) error {
```

Comment:
function should be independent of brick info, log etc. You can name it `LVSnapshot` and move caller to other file

Reply:
so that it can be reused by others. See `$SRC/plugins/device/deviceutils/utils.go`